WayMovies/Controller/BrowseViewController.swift

```diff
@@ -0,0 +1,399 @@
+                    collection.topAnchor.constraint(equalTo: scrollView.topAnchor),
+                    collection.leftAnchor.constraint(equalTo: view.leftAnchor),
+                    collection.rightAnchor.constraint(equalTo: view.rightAnchor),
+                    collection.heightAnchor.constraint(equalToConstant: collectionHeight)
+                ])
+            } else {
+                NSLayoutConstraint.activate([
+                    collection.topAnchor.constraint(equalTo: collectionList[index-1].bottomAnchor),
+                    collection.leftAnchor.constraint(equalTo: view.leftAnchor),
+                    collection.rightAnchor.constraint(equalTo: view.rightAnchor),
+                    collection.heightAnchor.constraint(equalToConstant: collectionHeight)
+                ])
+            }
+        }
+    }
+    
+    /**
+     Navigates to the favorites page of the item associated with the cell
+     */
+    @objc func viewFavorites(sender: UIBarButtonItem) {
+        navigationController?.pushViewController(FavoritesViewController(), animated: true)
+    }
+    
+//    @objc func refresh() {
+//        for collection in collectionList {
+//            collection.TVCollection.reloadData()
+//        }
+//        refreshControl.endRefreshing()
+//    }
+    
+}
+
+extension BrowseViewController: NavigationDelegate {
+    /**
+     Called when the cell is tapped to navigate to the details page of the item
+     */
+    func cellTapped(_ item: TVItem) {
+        navigationController?.pushViewController(DetailViewController(item: item), animated: true)
+    }
+}
+
+extension BrowseViewController: UITableViewDelegate, UITableViewDataSource {
+    /**
+     Returns the number of suggested categorized search options
+     */
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 3
+    }
+    
+    /**
+     Returns a tableview cell with a suggested categorized search
+     */
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cell")! //1.
+        cell.backgroundColor = .white
+        switch indexPath.row {
+        case 0:
+            cell.textLabel?.text = "\(searchBar.text ?? "") in Movies"
+        case 1:
+            cell.textLabel?.text = "\(searchBar.text ?? "") in TV Shows"
+        case 2:
+            cell.textLabel?.text = "\(searchBar.text ?? "") in People"
+        default:
+            cell.textLabel?.text = "Enter Search Terms"
+        }
+        return cell
+    }
+    
+    /**
+     Searches TMDB based on the categorized search option selected
+     */
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if searchBar.text == "" {
+            return
+        }
+        switch indexPath.row {
+        case 0:
+            Request.searchMovie(searchTerms: searchBar.text!) { [weak self] result in
```

Comment:
just nth - consider having a dedicated request manager / handler to make this view a little less smart